Apromore-Core-Components/Apromore-Portal/src/main/java/org/apromore/portal/security/helper/JwtHelper.java

```diff
@@ -0,0 +1,187 @@
+import java.text.SimpleDateFormat;
+import java.time.Duration;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.util.Date;
+
+public class JwtHelper {
+
+    public static final String JWT_KEY_SUBJECT_USERNAME = "sub";
+    public static final String STR_JWT_KEY_ISSUED_AT = "striat";
+    public static final String STR_JWT_EXPIRY_TIME = "strexp";
+    public static final Duration WEBAPP_SSO_SESSION_TIMEOUT = Duration.ofMinutes(30);
+
+    public static String readCookie(final HttpServletRequest httpServletRequest,
+                                    final String cookieName) {
+        final Cookie[] cookies = httpServletRequest.getCookies();
+
+        if (cookies != null) {
+            for (final Cookie cookie : cookies) {
+                if (cookie.getName().equals(cookieName)) {
+                    return cookie.getValue();
+                }
+            }
+        }
+
+        return null;
+    }
+
+    public static boolean isJwtExpired(final JWTClaimsSet jwtClaimsSet,
+                                       final String issuedAtStr,
+                                       final String expiredAtStr) {
+        long jwtExpiryEpoch;
+        if (expiredAtStr != null) {
+            jwtExpiryEpoch = Long.valueOf(expiredAtStr.substring(3));
+        } else { // Calculate relative
+            final Long jwtIssuedAt = Long.valueOf(issuedAtStr.substring(3));
+
+            jwtExpiryEpoch = calculateExpiryTime(jwtIssuedAt, WEBAPP_SSO_SESSION_TIMEOUT);
+        }
+
+        final Long nowEpoch = LocalDateTime.now().atZone(ZoneId.systemDefault()).toEpochSecond();
+
+        final boolean jwtExpired = (nowEpoch > jwtExpiryEpoch);
+
+        return jwtExpired;
+    }
+
+    public static JWTClaimsSet getClaimsSetFromJWT(final String jwtStr) throws Exception {
+        final JWT jwtParsed = JWTParser.parse(jwtStr);
+        final JWTClaimsSet jwtClaimsSet = jwtParsed.getJWTClaimsSet();
+
+        return jwtClaimsSet;
+    }
+
+    /**
+     * Return authenticated user if JWT token is not expired, otherwise <code>null</code>.
+     *
+     * @return
+     */
+    public static UserType userFromJwt(final ManagerService managerService, final String appAuthHeader) {
+        try {
+            final JWTClaimsSet jwtClaimsSet = JwtHelper.getClaimsSetFromJWT(appAuthHeader);
+
+            final String issuedAtStr = jwtClaimsSet.getStringClaim(JwtHelper.STR_JWT_KEY_ISSUED_AT);
+            final String expiryAtStr = jwtClaimsSet.getStringClaim(JwtHelper.STR_JWT_EXPIRY_TIME);
+
+            final boolean jwtExpired = JwtHelper.isJwtExpired(jwtClaimsSet, issuedAtStr, expiryAtStr);
+
+            if (!jwtExpired) {
+                final UserType authUserType =
+                        buildAuthenticatedUserFromJwt(jwtClaimsSet, managerService);
+
+                return authUserType;
+            } else {
+                return null;
+            }
+        } catch (final Exception pe) {
+            return null;
+        }
```

Comment:
Perhaps this exception should be logged?  (Perhaps the preceding JWT expiry as well?)

Reply:
Done (x2).